.github/workflows/docs.yml

```diff
@@ -0,0 +1,57 @@
+name: Deploy docs to Pages
+
+on:
+  push:
+    branches: ["main"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: 'pip'
+          cache-dependency-path: "docs/requirements.txt"
+      - name: Install dependencies
+        run: |
+          sudo apt install -y -qq doxygen
+      - name: Build Python
+        working-directory: python
+        run: |
+          ./tools/build_wheel.sh cp310
+          python -m pip install -r ../docs/requirements.txt
+          python -m pip install $(ls wheels/*.whl)
```

Comment:
Should we install the wheels first so their versions take precedence?

Reply:
Sure, can do